gulpfile.js

```diff
@@ -1,105 +1,55 @@
-const run = require("gulp-run-command").default;
+const { series, parallel } = require("gulp");
 
 // Include Our tasks.
 //
 // Each task is broken apart to it's own node module.
 // Check out the ./gulp-tasks directory for more.
-const {
-  unitTests,
-  sassTests,
-  a11y,
-  cover,
-} = require("./config/gulp/test");
-const {
-  cleanCSS,
-  cleanFonts,
-  cleanImages,
-  cleanJS,
-  cleanSass,
-} = require("./gulp-tasks/clean");
-const {
-  compileSass,
-  compileJS,
-  compileSprite,
-} = require("./gulp-tasks/compile");
-const {
-  copyVendor,
-  copySass,
-  copyImages,
-  copyFonts,
-  copyStyleguide,
-} = require("./gulp-tasks/copy");
+const { noCleanup, noTest } = require("./gulp-tasks/flags");
+const { buildSprite } = require("./gulp-tasks/svg-sprite");
+const { compileJS, typeCheck } = require("./gulp-tasks/javascript");
+const { unitTests, sassTests, a11y, cover } = require("./gulp-tasks/test");
 const { lintSass, lintJS } = require("./gulp-tasks/lint");
-const { moveFonts, movePatternCSS } = require("./gulp-tasks/move");
-const server = require("browser-sync").create();
-
-// Clean all directories.
-exports.clean = parallel(cleanCSS, cleanFonts, cleanImages, cleanJS, cleanSass);
+const { build } = require("./gulp-tasks/build");
+const { release } = require("./gulp-tasks/release");
+const { watch } = require("./gulp-tasks/watch");
+const { serve, buildPL, exitServer } = require("./gulp-tasks/serve");
+const { compileSass } = require("./gulp-tasks/sass");
+const { copyVendor } = require("./gulp-tasks/copy");
+const { cleanDist } = require("./gulp-tasks/clean");
 
-// Lint Sass
-exports.lintSass = parallel(lintSass);
+/**
+ * *Flags*
+ */
+exports.noTest = noTest;
+exports.noCleanup = noCleanup;
 
-// Lint JavaScript
-exports.lintJS = parallel(lintJS);
+/**
+ * *Clean tasks*
+ */
+exports.cleanDist = cleanDist;
 
-// Lint Sass and JavaScript
+/**
+ * *Lint tasks*
+ */
+exports.lintSass = lintSass;
+exports.lintJS = lintJS;
 exports.lint = parallel(lintSass, lintJS);
 
-// Compile Our Sass and JS
-exports.compile = parallel(
-  compileSass,
-  compileJS,
-  compileSprite,
-  moveFonts,
-  movePatternCSS
-);
-
-async function rebuildPL() {
-  return run("npm run pl:build --pattern")();
-}
-
-async function buildPL() {
-  return run("npm run pl:build")();
-}
-
-async function exitServer() {
-  return server.exit()
-}
-
 /**
- * Start browsersync server.
- * @param {function} done callback function.
- * @returns {undefined}
+ * *Test tasks*
+ * a11y: Accessibility tests; starts server, runs axe tests, and closes server.
+ * cover: Similar to unit tests.
+ * sassTests: Sass unit tests.
+ * unitTests: Component unit tests.
+ * test: Run all tests.
  */
-function serve(done) {
-  // See https://browsersync.io/docs/options for more options.
-  server.init({
-    server: ["./build/"],
-    notify: false,
-    open: false,
-    port: 3333,
-  });
-  done();
-}
-
 exports.a11y = series(serve, a11y, exitServer);
```

Comment:
this spins up a separate instance of browsersync server - if we already have one running - we probably don't need to spin up a new one if w have one